src/pages/home/report/EmojiPickerMenu/index.js

```diff
@@ -0,0 +1,132 @@
+        this.renderItem = this.renderItem.bind(this);
+
+        this.state = {
+            filteredEmojis: emojis,
+            headerIndices: this.unfilteredHeaderIndices,
+        };
+    }
+
+    /**
+     * Filter the entire list of emojis to only emojis that have the search term in their keywords
+     *
+     * @param {String} searchTerm
+     */
+    filterEmojis(searchTerm) {
+        const normalizedSearchTerm = searchTerm.toLowerCase();
+        if (normalizedSearchTerm === '') {
+            // There are no headers when searching, so we need to re-make them sticky when there is no search term
+            this.setState({filteredEmojis: emojis, headerIndices: this.unfilteredHeaderIndices});
+            return;
+        }
+        const newFilteredEmojiList = [];
+        _.each(emojis, (emoji) => {
+            if (emoji.header || emoji.code === CONST.EMOJI_SPACER) {
+                return;
+            }
+
+            if (_.find(emoji.keywords, keyword => keyword.includes(normalizedSearchTerm))) {
+                newFilteredEmojiList.push(emoji);
+            }
+        });
+
+        // Remove sticky header indices. There are no headers while searching and we don't want to make emojis sticky
+        this.setState({filteredEmojis: newFilteredEmojiList, headerIndices: []});
+    }
+
+    /**
+     * Given an emoji item object, render a component based on its type.
+     * Items with the code "SPACER" return nothing and are used to fill rows up to 8
+     * so that the sticky headers function properly
+     *
+     * @param {Object} item
+     * @returns {*}
+     */
+    renderItem({item}) {
+        if (item.code === CONST.EMOJI_SPACER) {
+            return null;
+        }
+
+        if (item.header) {
+            return (
+                <Text style={styles.emojiHeaderStyle}>
+                    {item.code}
+                </Text>
+            );
+        }
+
+        return (
+            <EmojiPickerMenuItem
+                onPress={this.props.onEmojiSelected}
+                emoji={item.code}
+            />
+        );
+    }
+
+    render() {
+        return (
+            <View style={styles.emojiPickerContainer}>
+                <View style={[styles.pt4, styles.ph4, styles.pb1]}>
+                    <TextInputFocusable
+                        textAlignVertical="top"
+                        placeholder="Search"
+                        placeholderTextColor={themeColors.textSupporting}
+                        onChangeText={this.filterEmojis}
+                        style={styles.textInput}
+                        defaultValue=""
+                        ref={el => this.searchInput = el}
+                    />
+                </View>
+                <FlatList
+                    data={this.state.filteredEmojis}
+                    renderItem={this.renderItem}
+                    keyExtractor={item => (`emoji_picker_${item.code}`)}
```

Comment:
NAB but are the extra parens necessary here?

```suggestion
                    keyExtractor={item => `emoji_picker_${item.code}`}
```